llama/generation.py

```diff
@@ -134,7 +134,7 @@ def __init__(self, model: Transformer, tokenizer: Tokenizer, device: torch.devic
             else:
                 self._generate_one_token_fn = torch.compile(
                     self._generate_one_token_fn,
-                    backend="openxla_eval",
+                    backend=dynamo,
```

Comment:
Is the "inductor" backend used for XLA:GPU case? I thought for PyTorch/XLA, "openxla*" is used for both XLA:TPU and XLA:GPU, am I misunderstanding?

Reply:
inductor is just used here to benchmark against  :-) We could easily set the default to openxla_eval -- either way one has to fiddle with the `USE_CUDA` environment variable. See above comment.

Reply:
Does that mean `USE_CUDA` and `dynamo` settings have dependency? What happens if the user sets `USE_CUDA=False` and `--dynamo=inductor`?

Reply:
That would use Inductor, but it would also break wrt the existing behaviour because we used to expect `USE_CUDA == Inductor`. I will try to remove that assumption.

Reply:
> Does that mean USE_CUDA and dynamo settings have dependency?
Indeed, that is currently the case. This PR so far makes things a bit better but doesn't fully disentangle that.

Reply:
Please see my comment below https://github.com/pytorch-tpu/llama/pull/47#issuecomment-1920150636